src/boot/efi/boot.c

```diff
@@ -1117,29 +1117,28 @@ static void config_add_entry(Config *config, ConfigEntry *entry) {
         config->entries[config->n_entries++] = entry;
 }
 
-static void config_entry_free(ConfigEntry *entry) {
-        if (!entry)
-                return;
+static ConfigEntry* config_entry_free(ConfigEntry *entry) {
+        if (entry) {
```

Comment:
this part doesn't look like an improvement. we usually prefer early exit over deep nesting.

i mean, I appreciate the conversion to return the type-safey NULL, but I think the large `if` block is not an improvement.

It's also not how we mostly do this.

hence please just do:

```c
if (!entry) 
        return NULL;

free(entry->…)
return mfree(entry);
```